src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -161,6 +173,128 @@ message InfoResponse {
     optional string sdk_language_version = 4;
 }
 
+// === run_task_allocations_session RPC === //
+
+message SerializedObjectID {
+    optional string id = 1;
+}
+
+message SerializedObjectManifest {
+    optional SerializedObjectID id = 1;
```

Comment:
How does the FE choose an ID?  And do these need to be strings?

Reply:
Both sides actually choose IDs themself. They don't have to be strings but strings give flexibility and readability for humans for free so I'd use them.